tests/components/overkiz/test_config_flow.py

```diff
@@ -0,0 +1,191 @@
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": TEST_EMAIL, "password": TEST_PASSWORD, "hub": TEST_HUB},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": error}
+
+
+async def test_abort_on_duplicate_entry(hass: HomeAssistant) -> None:
+    """Test config flow aborts Config Flow on duplicate entries."""
+    MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=TEST_GATEWAY_ID,
+        data={"username": TEST_EMAIL, "password": TEST_PASSWORD, "hub": TEST_HUB},
+    ).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("pyoverkiz.client.OverkizClient.login", return_value=True), patch(
+        "pyoverkiz.client.OverkizClient.get_gateways",
+        return_value=MOCK_GATEWAY_RESPONSE,
+    ), patch("homeassistant.components.overkiz.async_setup_entry", return_value=True):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": TEST_EMAIL, "password": TEST_PASSWORD},
+        )
+
+    assert result2["type"] == "abort"
+    assert result2["reason"] == "already_configured"
+
+
+async def test_allow_multiple_unique_entries(hass: HomeAssistant) -> None:
+    """Test config flow allows Config Flow unique entries."""
+    MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="test2@testdomain.com",
+        data={"username": "test2@testdomain.com", "password": TEST_PASSWORD},
+    ).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("pyoverkiz.client.OverkizClient.login", return_value=True), patch(
+        "pyoverkiz.client.OverkizClient.get_gateways",
+        return_value=MOCK_GATEWAY_RESPONSE,
+    ), patch("homeassistant.components.overkiz.async_setup_entry", return_value=True):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": TEST_EMAIL, "password": TEST_PASSWORD, "hub": TEST_HUB},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == TEST_EMAIL
+    assert result2["data"] == {
+        "username": TEST_EMAIL,
+        "password": TEST_PASSWORD,
+        "hub": TEST_HUB,
+    }
+
+
+async def test_dhcp_flow(hass: HomeAssistant) -> None:
+    """Test that DHCP discovery for new bridge works."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        data=dhcp.DhcpServiceInfo(
+            hostname="gateway-1234-5678-9123",
+            ip="192.168.1.4",
+            macaddress="F8811A000000",
+        ),
+        context={"source": config_entries.SOURCE_DHCP},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == config_entries.SOURCE_USER
```

Comment:
Please continue each test to a create entry, error or abort result. Each test should test a start point to one of those result types. There may be side effects so that later steps have different outcomes when starting from a different starting point.